providers/mongodb/fx.go

```diff
@@ -0,0 +1,75 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/fx"
+)
+
+func FXMongoDBConnection(cmd *cobra.Command, ctx context.Context, lc fx.Lifecycle) (*mongo.Database, error) {
+	var err error
+	host := viper.GetString(hostFlag)
+	if host == "" {
+		host, err = cmd.Flags().GetString(hostFlag)
+		if err != nil {
+			return nil, ErrCredentialsNotFound
+		}
+	}
+	port := viper.GetString(portFlag)
+	if port == "" {
+		port, err = cmd.Flags().GetString(portFlag)
+		if port == "" || err != nil {
+			port = "27017"
+		}
+	}
+	password := viper.GetString(passwordFlag)
+	if password == "" {
+		password, err = cmd.Flags().GetString(passwordFlag)
+		if err != nil {
+			return nil, ErrCredentialsNotFound
+		}
+	}
+	username := viper.GetString(userFlag)
+	if username == "" {
+		username, err = cmd.Flags().GetString(userFlag)
+		if err != nil {
+			return nil, ErrCredentialsNotFound
+		}
+	}
+	db := viper.GetString(dbFlag)
+	if db == "" {
+		db, err = cmd.Flags().GetString(dbFlag)
+		if err != nil {
+			return nil, ErrCredentialsNotFound
+		}
+	}
+	// Set client options
+	clientOptions := options.Client().ApplyURI(fmt.Sprintf("mongodb://%s:%s", host, port))
+	clientOptions = clientOptions.SetAuth(options.Credential{
```

Comment:
make sure to set the connection pool size or you're gonna drain your DB

for ex `clientOptions.SetMaxPoolSize(25)`